### PR TITLE
Apply Linting Only to Files Updated in git

### LINKED ISSUE #1270: Apply Linting Only to Files Updated in `git`
We need the ability to selectively adjust our linting rules so that they are eventually applied to our entire code base.

1. Linting rules are managed by the `.eslintrc.json` file.
1. They are applied to the entire code base at once.
1. If we change a rule from `warn` to `error` then every file in the code base will be checked and we have frequently seen our tests fail for dozens of files in hundreds of lines of code.

This is not acceptable.

We have been trying to more strictly lint the entire code base manually, but it has been very slow.

## At the user CLI
We need a way to edit `.eslintrc.json` with increasingly stricter linting rules in such a way that the tests are only applied to the changed files with every `git commit` and `git push` command. 

These `git` commands must fail if the linting fails.

## With Every Pull Request

We recently merged a Talawa-API pull request that checks each submitted file for linting compliance. This can be used as an example of how to run the test against individual files.

- https://github.com/PalisadoesFoundation/talawa-admin/pull/1242

You must make sure that this functionality applied for all PRs for this repository whenever the  `.eslintrc.json` file is updated

## Other

1. There may be ways to achieve the CLI goals by updating our existing Git Hooks
1. Your solution should preferably work within coding environments such as VScode
1. Please work with @anwersayeed and @tasneemkoushar on this issue
1. Related issues:
    - https://github.com/PalisadoesFoundation/talawa-api/issues/1548

## Screenshots

When complete, you should be able to convert all these  `.eslintrc.json` file lines containing `warn` to `error` and the tests will only be applied to the changed files in `git`

image

### PR DESCRIPTION
**Pull request by Shubh152**
What kind of change does this PR introduce?

feature-request

Issue Number:

Fixes #1270 

Did you add tests for your changes?

Snapshots/Videos:

If relevant, did you update the documentation?

Summary
Checks linting rules only for files that are staged for commit, will work with any changes to eslint or prettier configuration.
Uses lint-staged package with existing git hooks

Does this PR introduce a breaking change?

Other information

Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by Shubh152** [comment]
I closed the previous PR and created a new one. Please provide review @noman2002

**Comment by palisadoes** [comment]
1. Please don't close your PRs without discussing with the reviewers. 
2. We assign reviewers with each PR so it means that you increase our workload.

**Comment by Shubh152** [comment]
I am very sorry that it got closed automatically. I'll keep this in mind from now on.

**Comment by palisadoes** [comment]
@Nitya-Pasrija Please be aware of this PR

**Comment by Nitya-Pasrija** [comment]
I'll look into it

File: src/screens/OrgPost/OrgPost.tsx [hunk 1]
```diff
@@ -0,0 +1,501 @@
+import type { ChangeEvent } from 'react';
+import React, { useState, useEffect } from 'react';
+import { Search } from '@mui/icons-material';
```
Comment (palisadoes) [inline src/screens/OrgPost/OrgPost.tsx:3]: This file must not be included in this PR. It is out of scope

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,9 +1,10 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-npm run format:fix
-npm run lint:fix
+npm run lint-staged #runs the lint checks for staged files only, check .lintstagedrc.json for config
+# npm run format:fix
```
Comment (palisadoes) [inline .husky/pre-commit:5]: 1. Why was this done?
2. Are the references to `format:fix` and `lint:fix` still being used from the code base or documented? If not they should be removed.

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

File: src/screens/OrgPost/OrgPost.tsx (hunk 1 above)
Comment (Shubh152) [reply src/screens/OrgPost/OrgPost.tsx:3]: Can you please suggest how to remove this file from the PR?. I tried but could not do it.

File: .husky/pre-commit (hunk 2 above)
Comment (Shubh152) [reply .husky/pre-commit:5]: the 'lint-staged' script is using these commands indirectly (check .lintstagedrc.json). I had to remove them only from the git hooks so that only staged files are checked for linting.
